engine/src/main/java/org/camunda/bpm/engine/externaltask/LockedExternalTask.java

```diff
@@ -134,4 +136,19 @@
    */
   String getBusinessKey();
 
+  /**
+   * Returns a map of custom extension properties if the fetch instructions
+   * indicate to include extension properties.
+   * 
+   * If extension properties are included, the returned map contains any
+   * extension property that is defined in the model definition of the external
+   * task. If extension properties not included or no properties are defined for
+   * the external task, the map will be empty.
+   * 
+   * @return a potentially empty map of all defined custom extension properties
+   *         if the fetch-and-lock instructions include extension properties,
+   *         otherwise {@link Collections#emptyMap()}
```

Comment:
You explained the circumstances pretty well beforehand, no need for duplicated information I would say, the "never null" part is more interesting API-wise I would say
```suggestion
   * @return all defined custom extension properties, never <code>null</code>
```